packages/flutter_tools/lib/src/android/android_device.dart

```diff
@@ -622,18 +622,24 @@ final RegExp _kDeviceRegex = new RegExp(r'^(\S+)\s+(\S+)(.*)');
 /// [mockAdbOutput] is public for testing.
 List<AndroidDevice> getAdbDevices({ String mockAdbOutput }) {
   List<AndroidDevice> devices = <AndroidDevice>[];
-  List<String> output;
+  String text;
 
   if (mockAdbOutput == null) {
     String adbPath = getAdbPath(androidSdk);
     if (adbPath == null)
       return <AndroidDevice>[];
-    output = runSync(<String>[adbPath, 'devices', '-l']).trim().split('\n');
+    text = runSync(<String>[adbPath, 'devices', '-l']);
   } else {
-    output = mockAdbOutput.trim().split('\n');
+    text = mockAdbOutput;
   }
 
-  for (String line in output) {
+  // Check for error messages from adb
+  if (!text.contains('List of devices')) {
+    printError(text);
```

Comment:
Perhaps we should change the contract for `getAdbDevices()` to throw an exception if we have an error listing devices? As a get method, we should be able to call it w/o side effects. Instead of printing an error here, that would likely be the responsibility of a caller of the method.

I'm not sure who calls getAdbDevices, where the best place to print the error would be, and which other callers would have to be made aware that getAdbDevices could now throw...